packages/space-opera/src/components/materials_panel/reducer.ts

```diff
@@ -58,29 +60,30 @@ export interface AddTextureArgs {
  * Note: updateMaterial must NOT mutate the arg, but rather returns a new object
  */
 function setMaterialTexture(
-    state: State,
-    args: SetTextureArgs|undefined,
+    name: string,
+    args: SetTextureArgs,
+    materials: Material[],
     updateMaterial: (material: Material) => Material) {
-  if (!args)
-    return;
-  const materials = state.edits.materials;
   if (args.id >= materials.length || args.id < 0) {
     throw new Error('Given ID was out of bounds');
   }
-  if (args.textureId !== undefined &&
-      !state.edits.texturesById.has(args.textureId)) {
-    throw new Error(
-        `Tried to use a texture ID that does not exist: ${args.textureId}`);
-  }
+  // TODO: Add back in, but move reduxStore out of file, so pull textureIds from
```

Comment:
uncomments, and pretty sure this can never happen, considering the textures are pulled from the glb to begin with, kept just for posterity, to possibly be removed in the future